### DISCUSSION TITLE
Any optimization that I can do to improve `execute_field` performance?

### DISCUSSION THREAD

**Discussion by junyuanz1** [rmosolgo/graphql-ruby#4898] [Q&A] [answered]
our schema configuration looks like this:

1. we use lazy loader to load an object
2. and we need to return all the fields from that object

we noticed that the ruby-graphql has to call `execute_field` on every field one by one
since our field is pretty big, it ends up taking a long time to go thought all the fields

the graphql gem that i am using is very old `1.13.18` not sure whether newer version has any optimization around this or not.

**Comment by rmosolgo** [comment, marked as answer]
Hi! There's an internal wrapper called RawValue which GraphQL-Ruby inserts without modification into the response whenever it's found: 

https://github.com/rmosolgo/graphql-ruby/blob/24f797a09cf699e3b14d6ac60d4a72a53ec8f89b/lib/graphql/execution/interpreter/runtime.rb#L546-L549

You could try using that if you don't need any more GraphQL resolution. (But you have to handle _subfields, too, if there are any.) That's all I can think of! 

To dig in further, please open an issue with some profiling data, for example: https://graphql-ruby.org/testing/profiling.html
